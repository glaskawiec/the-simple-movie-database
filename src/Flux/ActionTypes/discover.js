export const DISCOVER_SET_PAGINATION = 'DISCOVER_SET_PAGINATION';
export const DISCOVER_SET_OPTIONS = 'DISCOVER_SET_OPTIONS';

