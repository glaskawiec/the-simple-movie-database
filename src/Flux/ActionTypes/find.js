export const FIND_SET_SEARCH_TEXT = 'FIND_SET_SEARCH_TEXT';
