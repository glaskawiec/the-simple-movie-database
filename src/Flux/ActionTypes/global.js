// eslint-disable-next-line import/prefer-default-export
export const GLOBAL_THEME_CHANGE = 'GLOBAL_THEME_CHANGE';
