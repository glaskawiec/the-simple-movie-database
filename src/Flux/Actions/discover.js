import {
  DISCOVER_SET_OPTIONS,
  DISCOVER_SET_PAGINATION,
} from '../ActionTypes/discover';

export const discoverSetPagination = pagination => ({
  type: DISCOVER_SET_PAGINATION,
  pagination,
});

export const discoverSetOptions = options => ({
  type: DISCOVER_SET_OPTIONS,
  options,
});
