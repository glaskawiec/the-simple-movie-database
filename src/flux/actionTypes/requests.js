export const REQUEST_IS_PENDING = 'REQUEST_IS_PENDING';
export const REQUEST_SUCCESS = 'REQUEST_SUCCESS';
export const REQUEST_ERROR = 'REQUEST_ERROR';
export const REQUEST_CLEAR = 'REQUEST_CLEAR';
