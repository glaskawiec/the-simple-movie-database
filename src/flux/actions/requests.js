import {
  REQUEST_CLEAR,
  REQUEST_FAILURE,
  REQUEST_IS_PENDING,
  REQUEST_SUCCESS,
} from '../actionTypes/requests';
import requestTheMovieDbApi from '../../utils/requestTheMovieDbApi';

const requestIsPending = id => ({
  type: REQUEST_IS_PENDING,
  id,
});

const requestSuccess = (id, responseData) => ({
  type: REQUEST_SUCCESS,
  id,
  responseData,
});

export const requestError = (id, error) => ({
  type: REQUEST_FAILURE,
  id,
  error,
});

export const requestClear = id => ({
  type: REQUEST_CLEAR,
  id,
});

export const requestApi = (id, request, afterSuccess = () => ({})) => async (dispatch) => {
  try {
    dispatch(requestIsPending(id));
    const response = await requestTheMovieDbApi(request);
    const parsedResponse = await response.json();
    dispatch(requestSuccess(id, parsedResponse));
    afterSuccess(parsedResponse);
  } catch (error) {
    dispatch(requestError(id, error));
  }
};
