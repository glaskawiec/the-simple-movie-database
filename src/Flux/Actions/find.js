import { FIND_SET_SEARCH_TEXT } from '../ActionTypes/find';

export const findSetSearchText = searchText => ({
  type: FIND_SET_SEARCH_TEXT,
  searchText,
});
