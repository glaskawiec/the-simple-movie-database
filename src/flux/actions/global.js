import { GLOBAL_THEME_CHANGE } from '../ActionTypes/global';

// eslint-disable-next-line import/prefer-default-export
export const globalThemeChange = theme => ({ type: GLOBAL_THEME_CHANGE, theme });
