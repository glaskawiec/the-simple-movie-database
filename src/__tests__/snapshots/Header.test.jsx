import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { MemoryRouter } from 'react-router-dom';
import { HouxProvider } from 'houx';
import themes from '../../themes';
import global from '../../flux/reducers/global';
import discover from '../../flux/reducers/discover';
import find from '../../flux/reducers/find';
import requests from '../../flux/reducers/requests';
import Header from '../../layout/header/Header';

const reducers = {
  global,
  discover,
  find,
  requests,
};

test('Header snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <HouxProvider reducers={reducers}>
          <ThemeProvider theme={themes.light}>
            <Header />
          </ThemeProvider>
        </HouxProvider>
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
