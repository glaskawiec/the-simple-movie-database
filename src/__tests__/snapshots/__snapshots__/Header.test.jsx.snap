// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div
  className="sc-bdVaJa cQKcmZ"
>
  <div
    className="sc-htpNat emOOYB"
  >
    <div
      className="sc-bwzfXH jjwbUn"
      onClick={[Function]}
    >
      tsmdb
    </div>
    <ul
      className="sc-gZMcBi gTagNS"
    >
      <li
        className="sc-iwsKbI huFlvw"
        onClick={[Function]}
      >
        discover
      </li>
      <li
        className="sc-iwsKbI huFlvw"
        onClick={[Function]}
      >
        find
      </li>
    </ul>
    <div
      className="sc-jTzLTM koDlOu"
      onClick={[Function]}
    >
      <div
        className="sc-fjdhpX bUtqfO"
      >
        <div
          className="sc-jzJRlG dAerCH"
        >
          <svg
            viewBox="-461.4 274.9 16 16"
          >
            <path
              d="M-456.1,285.7c3.3,0,5.9-2.6,5.9-5.9c0-1.4-0.5-2.8-1.4-3.8c2.7,0.9,4.6,3.3,4.6,6.3c0,3.7-3,6.6-6.6,6.6c-2.9,0-5.4-1.9-6.3-4.6C-458.8,285.1-457.5,285.7-456.1,285.7z"
              fill="#FFD93B"
            />
            <path
              d="M-449.5,277.1c0.9,1.1,1.4,2.5,1.4,4.1c0,3.7-3,6.6-6.6,6.6c-1.5,0-3-0.5-4.1-1.4c1.2,1.5,3.1,2.5,5.2,2.5c3.7,0,6.6-3,6.6-6.6C-447,280.2-448,278.3-449.5,277.1z"
              fill="#F4C534"
            />
          </svg>
        </div>
        <div
          className="sc-VigVT dcGKWJ"
        >
          <svg
            viewBox="-461.4 274.4 16 16"
          >
            <path
              d="M-445.9,283.8l-1.6-1.4l1.6-1.5c0.1-0.1,0.1-0.1,0.1-0.2s-0.1-0.1-0.2-0.2l-2.1-0.6l0.8-2c0-0.1,0-0.2,0-0.2c-0.1-0.1-0.1-0.1-0.2-0.1l-2.2,0.3l-0.2-2.2c0-0.1-0.1-0.1-0.1-0.2c-0.1,0-0.2,0-0.2,0l-1.8,1.2l-1.1-1.9c0-0.1-0.1-0.1-0.2-0.1s-0.2,0-0.2,0.1l-1.3,2.2l-2.2-1.2c-0.1,0-0.2,0-0.2,0c-0.1,0-0.1,0.1-0.1,0.2l0,2.2l-2.2-0.2c-0.1,0-0.2,0-0.2,0.1c0,0.1-0.1,0.2,0,0.2l0.9,2l-2,0.8c-0.1,0-0.1,0.1-0.1,0.2c0,0.1,0,0.2,0.1,0.2l1.7,1.4l-1.5,1.6c-0.1,0.1-0.1,0.1-0.1,0.2c0,0.1,0.1,0.1,0.2,0.2l2.1,0.5l-0.7,2.1c0,0.1,0,0.2,0,0.2c0.1,0.1,0.1,0.1,0.2,0.1l2.1-0.4l0.3,2.2c0,0.1,0.1,0.1,0.1,0.2c0.1,0,0.2,0,0.2,0l1.7-1.3l1.2,1.8c0,0.1,0.1,0.1,0.2,0.1c0,0,0,0,0,0c0.1,0,0.2,0,0.2-0.1l1-1.9l1.8,1.2c0.1,0,0.2,0,0.2,0c0.1,0,0.1-0.1,0.1-0.2l0.1-2.2l2.2,0.3c0.1,0,0.2,0,0.2-0.1c0-0.1,0.1-0.1,0-0.2l-0.9-2l2.1-0.7c0.1,0,0.1-0.1,0.1-0.2C-445.8,283.9-445.9,283.8-445.9,283.8z"
              fill="#FFA600"
            />
            <ellipse
              cx="-453.5"
              cy="282.4"
              fill="#FFDB2D"
              rx="4.6"
              ry="4.6"
            />
            <path
              d="M-453.5,277.8c-0.1,0-0.2,0-0.3,0c2.4,0.2,4.3,2.2,4.3,4.6c0,2.4-1.9,4.4-4.3,4.6c0.1,0,0.2,0,0.3,0c2.5,0,4.6-2.1,4.6-4.6C-448.9,279.8-450.9,277.8-453.5,277.8z"
              fill="#FFCA00"
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-gqjmRU gdzbDt"
      />
    </div>
  </div>
</div>
`;
